addons/hr_holidays/models/hr_leave.py

```diff
@@ -529,7 +529,8 @@ def _get_durations(self, check_leave_type=True, resource_calendar=None):
         employee_leaves = self.filtered('employee_id')
         employees_by_dates_calendar = defaultdict(lambda: self.env['hr.employee'])
         for leave in employee_leaves:
-            employees_by_dates_calendar[(leave.date_from, leave.date_to, resource_calendar or leave.resource_calendar_id)] += leave.employee_id
+            if leave.date_from and leave.date_to and (resource_calendar or leave.resource_calendar_id):
+                employees_by_dates_calendar[(leave.date_from, leave.date_to, resource_calendar or leave.resource_calendar_id)] += leave.employee_id
```

Comment:
If I'm not mistaken, the check on the calendar isn't necessary as the `resource_calendar_id` of the leave is always set.
Moreover, since we're in a loop, we can use continue inside the condition to avoid over-indenting a line that is already exceeding the maximum recommended line length :nerd_face: 
```suggestion
            if not leave.date_from or not leave.date_to:
                continue
            employees_by_dates_calendar[(leave.date_from, leave.date_to, resource_calendar or leave.resource_calendar_id)] += leave.employee_id
```